compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -604,6 +605,20 @@ impl<'a> Parser<'a> {
         self.token.is_keyword(kw)
     }
 
+    fn check_keyword_case(&mut self, kw: Symbol, case: Case) -> bool {
+        if self.check_keyword(kw) {
+            return true;
+        }
+
+        if case == Case::Insensitive
+        && let Some((ident, /* is_raw */ false)) = self.token.ident()
+        && ident.as_str().to_lowercase() == kw.as_str().to_lowercase() {
+            true
+        } else {
+            false
+        }
```

Comment:
I don't see any *obvious* reason for the slow down, but can you try putting up  PRs that revert some of these changes independently? Let's try to find if there's a single method that's introducing most of the perf regression (and it very well could be that it's just worse cache locality due to change in method inlining or arguments or something "silly" like that).